tests/sentry/issues/auto_source_code_config/test_process_event.py

```diff
@@ -39,37 +39,56 @@ def setUp(self) -> None:
             metadata={"domain_name": f"{self.domain_name}/test-org"},
         )
 
-    def create_event(self, frames: Sequence[Mapping[str, str | bool]], platform: str) -> Event:
+    def create_event(self, frames: Sequence[Mapping[str, str | bool]], platform: str) -> GroupEvent:
         """Helper function to prevent creating an event without a platform."""
         test_data = {"platform": platform, "stacktrace": {"frames": frames}}
-        return self.store_event(data=test_data, project_id=self.project.id)
+        # XXX: In the future fix store_event to return the correct type
+        return cast(GroupEvent, self.store_event(data=test_data, project_id=self.project.id))
 
     def _get_trees_for_org(self, files: Sequence[str]) -> dict[str, RepoTree]:
         return {"test-org/repo": RepoTree(RepoAndBranch("test-org/repo", "master"), files)}
 
     def _process_and_assert_code_mapping(
-        self, files: Sequence[str], stack_root: str, source_root: str
+        self,
+        *,  # Force keyword arguments
+        repo_files: Sequence[str],
+        frames: Sequence[Mapping[str, str | bool]],
+        platform: str,
+        expected_stack_root: str,
+        expected_source_root: str,
     ) -> None:
         with (
-            patch(GET_TREES_FOR_ORG, return_value=self._get_trees_for_org(files)),
+            patch(GET_TREES_FOR_ORG, return_value=self._get_trees_for_org(repo_files)),
             patch("sentry.utils.metrics.incr") as mock_incr,
         ):
-            process_event(self.project.id, self.event.group_id, self.event.event_id)
+            event = self.create_event(frames, platform)
```

Comment:
This event creation used to happen here:
<img width="462" alt="image" src="https://github.com/user-attachments/assets/36a1ad83-d230-4309-ac13-12e0f49f2f2c" />